### PR TITLE
Insights advisor filtering BDD

### PR DESCRIPTION
**Pull request by tisnik**

### CODE REVIEW

**Review by dpensi — APPROVED** [review]

File: features/Insights_Advisor_recommendations_filtering.feature [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+      And table with several columns should be displayed
+          | Column name |
+          | Name        |
+          | Added       |
+          | Total risk  |
+          | Clusters    |
+      And that table should contain exactly 1 row
+          | Name        | Added       | Total risk | Clusters |
+          | Bug12345    | 10 days ago | Important  | 1        |
+      And one filter needs to be preselenced
```
[the remark this answers is not in the document]
Comment (epapbak) [reply features/Insights_Advisor_recommendations_filtering.feature:45]: careful, it's repeated in other tests =)

[the remark this answers is not in the document]
Comment (tisnik) [reply features/Insights_Advisor_recommendations_filtering.feature]: Fixed everywhere

File: features/Insights_Advisor_recommendations_filtering.feature [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+          | Filter name       | Tag         |
+          | Clusters impacted | 1 or more   |
+     When user clicks on "x" icon displayed on "1 or more" filter tag
+     Then table with advisor recommendation should contain recommendations names with 0 clusters
+          | Name        | Added       | Total risk | Clusters |
+          | Abc12345    | 10 days ago | Important  | 0        |
+          | Bug12345    | 10 days ago | Important  | 1        |
+          | Xyz12345    | 10 days ago | Moderate   | 0        |
+      And filter widget should consists of filter names pull-down menu and filter search input field
+     When user selects "Total risk" in filter names pull-down menu
```
Comment (epapbak) [inline features/Insights_Advisor_recommendations_filtering.feature:245]: We can add here:
```
Then search input field becomes a pull-down menu with several options
  | Option name |
  | Critical           |
  | Important      |
  | Moderate      |
  | Low                |
```

Comment (epapbak) [reply features/Insights_Advisor_recommendations_filtering.feature:245]: If you do this change, the "And user select "Moderate" from the rigth pull-down menu" would become "When user selects..."

File: features/Insights_Advisor_recommendations_filtering.feature [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+          | Filter name       | Tag         |
+          | Clusters impacted | 1 or more   |
+     When user clicks on "x" icon displayed on "1 or more" filter tag
+     Then table with advisor recommendation should contain recommendations names with 0 clusters
+          | Name        | Added       | Total risk | Clusters |
+          | Abc12345    | 10 days ago | Important  | 0        |
+          | Bug12345    | 10 days ago | Important  | 1        |
+          | Xyz12345    | 10 days ago | Moderate   | 0        |
+      And filter widget should consists of filter names pull-down menu and filter search input field
+     When user selects "Clusters impacted" in filter names pull-down menu
```
Comment (epapbak) [inline features/Insights_Advisor_recommendations_filtering.feature:328]: same comment as previous test, if applied

File: features/Insights_Advisor_recommendations_filtering.feature [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+          | Name        |
+          | Added       |
+          | Total risk  |
+          | Clusters    |
+      And that table should contain three rows
+          | Name        | Added       | Total risk | Clusters |
+          | Abc12345    | 10 days ago | Important  | 0        |
+          | Bug12345    | 10 days ago | Important  | 1        |
+          | Xyz12345    | 10 days ago | Moderate   | 0        |
+
```
Comment (epapbak) [inline features/Insights_Advisor_recommendations_filtering.feature:354]: Why not add "user select "None" from the rigth pull-down menu" case (quick copy-paste, more coverage)

File: features/Insights_Advisor_recommendations_filtering.feature (hunk 2 above)
Comment (tisnik) [reply features/Insights_Advisor_recommendations_filtering.feature:245]: added

**Review by epapbak — COMMENTED** [review]
Awesome BDDs!

File: features/Insights_Advisor_recommendations_filtering.feature (hunk 4 above)
Comment (tisnik) [reply features/Insights_Advisor_recommendations_filtering.feature:360]: added

File: features/Insights_Advisor_recommendations_filtering.feature (hunk 3 above)
Comment (tisnik) [reply features/Insights_Advisor_recommendations_filtering.feature:334]: added

File: features/Insights_Advisor_recommendations_filtering.feature (hunk 4 above)
Comment (tisnik) [reply features/Insights_Advisor_recommendations_filtering.feature:360]: added too :)

**Pull request merged**